STEER/AOD/AliAODEvent.cxx

```diff
@@ -733,17 +733,46 @@ void AliAODEvent::ReadFromTree(TTree *tree, Option_t* opt /*= ""*/)
     // Try to find AliAODEvent
   AliAODEvent *aodEvent = 0;
   aodEvent = (AliAODEvent*)tree->GetTree()->GetUserInfo()->FindObject("AliAODEvent");
+  
+  TList* friendL;
   if(aodEvent){
     // This event is connected to the tree by definition, just say so
     aodEvent->SetConnected();
       // Check if already connected to tree
     TList* connectedList = (TList*) (tree->GetUserInfo()->FindObject("AODObjectsConnectedToTree"));
     if (connectedList && (!strcmp(opt, "reconnect"))) {
+		
         // If connected use the connected list of objects
         if (fAODObjects != connectedList) {
            delete fAODObjects;
            fAODObjects = connectedList;
         }   
+        
+        friendL = tree->GetTree()->GetListOfFriends();
+		if (friendL) 
+		{
+		  // set the branch addresses
+		TIter next(fAODObjects);
+		TNamed *el;
+		while((el=(TNamed*)next())){
+		  TString bname(el->GetName());
+			// check if branch exists under this Name
+		  TBranch *br = tree->GetTree()->GetBranch(bname.Data());
+		  if(br){
+			tree->SetBranchAddress(bname.Data(),fAODObjects->GetObjectRef(el));
+		  } else {
+			br = tree->GetBranch(Form("%s.",bname.Data()));
+			if(br){
+			  tree->SetBranchAddress(Form("%s.",bname.Data()),fAODObjects->GetObjectRef(el));
+			}
+			else{
+			  printf("%s %d AliAODEvent::ReadFromTree() No Branch found with Name %s. \n",
+					 (char*)__FILE__,__LINE__,bname.Data());
+			}	
+		  }
+		}
```

Comment:
It seems to me that this block was executed before in all cases (a bit below), so why adding it again will change something?

Reply:
After loading the first tree in AliAODInputHandler::Init, now instead of not extracting any photon conversion candidates, it now extracted them only for the first tree. So it seemed to me that only the branch addresses of the first tree are set properly. Setting them again for every subsequent tree fixed this issue for me.

Reply:
Maybe, but I see that the lines that you added  are the same as the ones below at [lines 823-838]( https://github.com/alisw/AliRoot/blob/d755faa53c655315b4e58f6eb0ba61524eff4987/STEER/AOD/AliAODEvent.cxx#L823-L838) so this code was called anyway, now it is called twice, or am I missing something?

Reply:
It is indeed the same code. But if I see correctly, that part is only executed for the first tree because of the `if (connectedList && (!strcmp(opt, "reconnect")))` statement. So for any subsequent tree, the connected list exists but the branch addresses are not set correctly, or are they?